content/en/logs/log_collection/javascript.md

```diff
@@ -0,0 +1,158 @@
+  text: "Perform Log Analytics"
+- link: "logs/faq/log-collection-troubleshooting-guide"
+  tag: "FAQ"
+  text: "Log Collection Troubleshooting Guide"
+---
+
+
+
+Send logs to Datadog from web browsers or other Javascript client thanks to the Datadog Client-side JavaScript logging library.
+
+With the `datadog-logs` library you are able to send log directory to Datadog from JS clients and leverage the following features:
+
+* Use the library as a logger. Everything is forwarded to Datadog as JSON documents.
+* Add `context` and extra attributes.
+* Forward every JavaScript errors.
+* Forward JavaScript's console logs.
+* Track real client IP address and useragent.
+* Automatic bulk posts.
+
+## Get the Public API Key
+
+**Public API Keys are in Private Beta** 
+
+For security reasons, [API keys][2] cannot be used as they would be exposed in the Javascript code. To collect Logs from web browsers, a [Public API keys][3] must be used.
+To manage your Public API keys, go to your [Datadog API configuration page][4] in the `Public API Key` section as shown here:
+
+{{< img src="logs/log_collection/public_key.png" style="width:80%;" alt="Public API Keys" responsive="true" >}}
+
+Read our [Public API keys documentation][3] to learn more about the restrictions that applies.
+
+## Configure the Javascript logger
+
+The following parameters can be used to configure the library to send logs to Datadog:
+
+* Set `isCollectingError` to false to turn off the automatic JS and console error collection
+* Use `addGlobalContext` to add JSON attribute to all the generated logs
+* Set `publicApiKey` to the value of the Public api_key (**only public api_key can be used in this library**)
+
+{{< tabs >}}
+{{% tab "US" %}}
+
+```
+<html>
+  <head>
+    <title>Example to send logs to Datadog</title>
+    <script type="text/javascript" src="https://www.datadoghq-browser-agent.com/datadog-logs-us.js"></script>
+    <script>
+      // Set your Public API key
+      Datadog.init({
+        publicApiKey: '<PUBLIC_API_KEY>',
+        isCollectingError: true,
+    });
+
+      // OPTIONAL
+      // add global metadata attributes
+      Datadog.addGlobalContext({'session_id': '1234'});
+    </script>
+    ...
+  </head>
+...
+</html>
+```
+
+{{% /tab %}}
+{{% tab "EU" %}}
+
+```
+<html>
+  <head>
+    <title>Example to send logs to Datadog</title>
+    <script type="text/javascript" src="https://www.datadoghq-browser-agent.com/datadog-logs-eu.js"></script>
+    <script>
+      // Set your Public API key
+      Datadog.init({
+        publicApiKey: '<PUBLIC_API_KEY>',
+        isCollectingError: true,
+    });
+
+      // OPTIONAL
+      // add global metadata attributes
+      Datadog.addGlobalContext({'session_id': '1234'});
+    </script>
+    ...
+  </head>
+...
+</html>
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Send a custom log entry
+
+It is then possible to send custom log entries thanks to the `log` function:
+
+```
+Datadog.log(<message>,<json_parameter>,<severity>)
```

Comment:
```suggestion
Datadog.log(<MESSAGE>,<JSON_ATTRIBUTES>,<SEVERITY>)
```

Reply:
@NBParis, just to confirm, JSON_ATTRIBUTES can be a dump of a JSON object that will be send alongside of the message?